### PR TITLE
Support `url.query` from OTel stable HTTP semconv

### LINKED ISSUE #3580: Azure ApplicationInsights stopped reporting Query paramters in Request traces - With 3.5.0
### Expected behavior
With previous versions the complete http path along with query parameter was reported in the HTTP request span in the application insights view. 

### Actual behavior
With new 3.5.0 version the HTTP request span only show the reduced path of the request without query parameter. 

### System information
Please provide the following information:
 - SDK Version: 3.5.0
 - OS type and version: N/A
 - Application Server type and version (if applicable): N/A
 - Using spring-boot? 3.2.2
 - Additional relevant libraries (with version, if applicable): N/A
 
 Looks like the semantic change might have created this issue - https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/migration-guide.md#summary-of-changes

### PR DESCRIPTION
**Pull request by trask**
Resolves #3580

### CODE REVIEW

File: smoke-tests/apps/HttpServerQueryString/src/smokeTest/resources/logback-test.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jeanbisutti) [inline smoke-tests/apps/HttpServerQueryString/src/smokeTest/resources/logback-test.xml:1]: Not sure this file is needed to display the Logback logs on the console. From what I have seen in the Logback code, if no appender is defined then the ConsoleAppender should be added by default.

File: smoke-tests/apps/HttpServerQueryString/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerQueryStringTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+
+  @Test
+  @TargetUri("/test?query")
+  void queryTest() throws Exception {
+    Telemetry telemetry = testing.getTelemetry(0);
+
+    assertThat(telemetry.rd.getName()).isEqualTo("GET /HttpServerQueryString/test");
+    assertThat(telemetry.rd.getUrl())
+        .matches("http://localhost:[0-9]+/HttpServerQueryString/test\\?query");
```
Comment (jeanbisutti) [inline smoke-tests/apps/HttpServerQueryString/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerQueryStringTest.java:33]: Related to the exporter code, is there a test with no port defined?

File: smoke-tests/apps/HttpServerQueryString/src/smokeTest/resources/logback-test.xml (hunk 1 above)
Comment (trask) [reply smoke-tests/apps/HttpServerQueryString/src/smokeTest/resources/logback-test.xml:1]: I think it's used to change the root level to warn, but not sure (it's included in all of the smoke tests). We could try it out separate from this PR and remove it from all smoke tests if it's not needed.

Comment (trask) [reply smoke-tests/apps/HttpServerQueryString/src/smokeTest/resources/logback-test.xml:1]: I _think_ this file is needed for setting the default root level below

File: smoke-tests/apps/HttpServerQueryString/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerQueryStringTest.java (hunk 2 above)
Comment (trask) [reply smoke-tests/apps/HttpServerQueryString/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerQueryStringTest.java:33]: good point, working on it

Comment (trask) [reply smoke-tests/apps/HttpServerQueryString/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerQueryStringTest.java:33]: I added a test for default http port, would be much more work to test default https port

File: smoke-tests/apps/HttpServer/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerDefaultPortTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  static final SmokeTestExtension testing =
+      SmokeTestExtension.builder().setAppServerPort(80).build();
+
+  @Test
+  @TargetUri("/test?query")
+  void queryTest() throws Exception {
+    Telemetry telemetry = testing.getTelemetry(0);
+
+    assertThat(telemetry.rd.getName()).isEqualTo("GET /HttpServer/test");
+    assertThat(telemetry.rd.getUrl()).isEqualTo("http://localhost/HttpServer/test?query");
```
Comment (heyams) [inline smoke-tests/apps/HttpServer/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerDefaultPortTest.java:35]: should the url contains port 80?

File: smoke-tests/apps/HttpServer/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerQueryStringTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+
+  @Test
+  @TargetUri("/test?query")
+  void queryTest() throws Exception {
+    Telemetry telemetry = testing.getTelemetry(0);
+
+    assertThat(telemetry.rd.getName()).isEqualTo("GET /HttpServer/test");
+    assertThat(telemetry.rd.getUrl()).matches("http://localhost:[0-9]+/HttpServer/test\\?query");
```
Comment (heyams) [inline smoke-tests/apps/HttpServer/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpServerQueryStringTest.java:32]: this considers the default random port.

File: smoke-tests/framework/src/main/java/com/microsoft/applicationinsights/smoketest/SmokeTestExtension.java [hunk 5]
```diff
@@ -242,11 +249,17 @@ public void beforeEach(ExtensionContext context) throws Exception {
 
   protected String getBaseUrl() {
     String appContext = getAppContext();
-    if (appContext.isEmpty()) {
-      return "http://localhost:" + appServerPort;
-    } else {
-      return "http://localhost:" + appServerPort + "/" + appContext;
+    StringBuilder sb = new StringBuilder();
+    sb.append("http://localhost");
+    if (appServerPort != 80) {
```
Comment (heyams) [inline smoke-tests/framework/src/main/java/com/microsoft/applicationinsights/smoketest/SmokeTestExtension.java:254]: i see, you only append port when it's not 80 (custom port). can we do that opposite instead... default, we don't append, but when it's 80, append?

**Review by heyams — APPROVED** [review]

**Pull request merged**